lib/compat/wordpress-6.0/class-wp-webfonts-provider-local.php

```diff
@@ -227,9 +227,9 @@ private function compile_src( $font_family, array $value ) {
 		$src = "local($font_family)";
 
 		foreach ( $value as $item ) {
-
-			if ( 0 === strpos( $item['url'], get_site_url() ) ) {
-				$item['url'] = wp_make_link_relative( $item['url'] );
+			if ( 0 === strpos( $item['url'], 'file:./' ) ) {
+				$absolute_path_to_url = get_stylesheet_directory_uri() . '/' . str_replace( 'file:./', '', $item['url'] );
```

Comment:
A couple of things here:

- `get_stylesheet_directory_uri()` limits this functionality to only themes - meaning plugins cannot use the `file:./` notation. Is this intended? Is `file:./` notation now only available for themes, meaning plugins are required to register local web fonts with absolute path to where the file(s) exist in the plugin?

- Micro-optimization: `get_site_url()` and/or `get_stylesheet_directory_uri()` can be cached within the provider object (either through memoization within this method or as a property) and then the value used here with the `foreach` loop.

- `0 === strpos( $item['url'], 'file:./' )` can be replaced with the more readable `str_starts_with()` function that was introduced in WP 5.9.

Reply:
>Is this intended? [...] meaning plugins are required to register local web fonts with absolute path to where the file(s) exist in the plugin

Yes -- for now. In `theme.json` you cannot specify a dynamic path that'd get resolved relative to the theme folder, that's why this notation exists.

In plugins/PHP realm, you can always pass the full URL built from the helper methods. We might make it work for plugins and themes in the future. This is just moving existing code around, not adding anything there.